feature_details.md

```diff
@@ -0,0 +1,174 @@
+
+#### Hierarchical Properties
+The web platform has multiple concepts which define the coordinate space in which the animatable properties referenced above should be interpolated : [Containing Block](https://developer.mozilla.org/en-US/docs/Web/CSS/Containing_block), [3D rendering context](https://drafts.csswg.org/css-transforms-2/#3d-rendering-contexts); the order in which the elements are stacked : [Stacking context](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Positioning/Understanding_z_index/The_stacking_context); and the element to which an effect like backdrop-filter is applied : [Backdrop Root](https://drafts.fxtf.org/filter-effects-2/#BackdropRoot). All these concepts are defined based on where an element fits in the DOM hierarchy. These are tricky to reason about in the context of a transition because :
+
+* It is desirable to change the ancestor for these concepts during the transition. For example, an element may be clipped by the viewport instead of an ancestor element as it animates to exit the screen.
+
+* Paired transitions are executed by creating a combined image for elements in the two trees on which the animated properties are applied. Should the transition switch to using a parent stacking context for the shared element in the new Document when a transition starts?
+
+The following steps define the behaviour of how content from the two Documents is drawn together during the transition. The aim is to avoid an element’s visual representation abruptly changing at transition start/end for common use-cases :
+
+* We assume that the dimensions for the initial containing block for both Documents is the same. If this changes, for example because the user resizes the browser window before the transition finishes, the transition is aborted.
+
+* During the transition, the UA creates a new root element (hidden from script), referred to as `TransitionRoot` which establishes the root stacking context and containing block. The background color for the `TransitionRoot` is the same as the background color for the root element in the new Document. By default, all shared elements (including root elements from both Documents) are direct descendants of the `TransitionRoot`.
+
+* TODO : Clarify the details of how properties are computed and animated during the transition.
+
+* We also need to define the order in which elements which share a stacking context are rendered. The current proposal allows the developer to explicitly control this by providing an ordered list (back-to-front) of `ElementTransition` in the `handleTransition.initiateTransition` API. But it’s unclear if this control is unnecessary complexity. The common use-case would be to maintain the same visual order for elements within the previous and new Document to avoid abrupt changes when the transition starts and ends.
```

Comment:
I keep coming back to the fact that this is a potentially 300ms transition where some buttons and logos fly around. I agree that the relative paint order of element is something that needs to be defined -- perhaps it can inferred from the relative paint order of either the incoming or outgoing pages -- but I'm against having this control in developer's hands, because at the end of the day I don't think it matters and it only complicates the API. However, I'm happy to be proven wrong here.

Reply:
I agree that this doesn't need to be a developer control but we'll need to think through a few cases to define the behaviour here : 

a) The order in which root elements draw, old or new on top. The current implementation uses a single enum to capture that behaviour. But in this proposal I was trying to define them as 2 single transition types : {new:none, old: slide-left}, where old  content draws on top and slides left to reveal the new content.

b) Assuming shared elements within a document draw in the same order. Is it all old elements stacked behind new ones or vice versa.

c) What do we do for paired elements.